src/sql/src/plan/statement/ddl.rs

```diff
@@ -3410,18 +3414,20 @@ pub fn plan_drop_cluster_replica(
         // Check to see if name exists
         if instance.replica_names().contains(&replica_name) {
             // Check if we have an item that depends on the replica's logs
-            let log_ids = instance.replica_logs(&replica_name).unwrap();
-            for id in log_ids {
-                let log_item = scx.catalog.get_item(&id);
-                for id in log_item.used_by() {
-                    let dep = scx.catalog.get_item(id);
-                    if dependency_prevents_drop(ObjectType::Source, dep) {
-                        sql_bail!(
-                            "cannot drop replica {} of cluster {}: still depended upon by catalog item '{}'",
-                            replica_name.quoted(),
-                            instance.name().quoted(),
-                            scx.catalog.resolve_full_name(dep.name())
-                        );
+            if !cascade {
+                let log_ids = instance.replica_logs(&replica_name).unwrap();
+                for id in log_ids {
+                    let log_item = scx.catalog.get_item(&id);
+                    for id in log_item.used_by() {
+                        let dep = scx.catalog.get_item(id);
+                        if dependency_prevents_drop(ObjectType::Source, dep) {
```

Comment:
This should be `ObjectType::ClusterReplica`, I believe

Reply:
No this is correct, since we do the one-hop away items from ClusterReplica, it should be Sources (this is how the intro sources are currently presented)